### PR TITLE
fix(curriculum): Update tests, remove reference to self-closing element

### PR DESCRIPTION
**Pull request by lasjorg**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Related #55342

I was just going to remove the reference to self-closing element, but I started looking at the test and wanted to update them a bit. I have regrets, lol.

Some of the extra element placement tests are to avoid issues with browser magic. E.g. the body element placement test, is because if you add the body element before the head element the browser does its magic, and you end up with the content of the head element inside the body element, but the position of the two elements are corrected.

I'm not sure the tests will catch all possible errors, but it should be a little better now.

---

Note to self, `\n` does not work on Linux/WSL. Spent way to much time on a regex that worked on Windows (VS Code and on regexr.com) but kept failing the tests in WSL. I should have known better.

### CODE REVIEW

File: curriculum/challenges/english/14-responsive-web-design-22/learn-css-animation-by-building-a-ferris-wheel/6140c7e645d8e905819f1dd4.md [hunk 1]
```diff
@@ -15,88 +15,100 @@ Set the `title` to `Ferris Wheel`.
 
 # --hints--
 
-Your code should contain the `DOCTYPE` reference.
+Your code should contain the `DOCTYPE` preamble.
```
Comment (Sembauke) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-css-animation-by-building-a-ferris-wheel/6140c7e645d8e905819f1dd4.md:18]:
```suggestion
Your code should contain the `DOCTYPE` declaration.
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-css-animation-by-building-a-ferris-wheel/6140c7e645d8e905819f1dd4.md [hunk 2]
```diff
@@ -15,88 +15,100 @@ Set the `title` to `Ferris Wheel`.
...
 
 ```js
-assert(code.match(/<!DOCTYPE/gi));
+assert.match(code, /<!DOCTYPE/gi);
 ```
 
-You should include a space after the `DOCTYPE` reference.
+You should include a space after the `DOCTYPE` preamble.
```
Comment (Sembauke) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-css-animation-by-building-a-ferris-wheel/6140c7e645d8e905819f1dd4.md:24]:
```suggestion
You should include a space after the `DOCTYPE` declaration.
```

**Review by Sembauke — CHANGES_REQUESTED** [review]
I never heard of preamble before. Can we write it with a word more commonly used?

**Comment by lasjorg** [comment]
The specs call it a preamble.

https://developer.mozilla.org/en-US/docs/Glossary/Doctype

https://html.spec.whatwg.org/multipage/syntax.html#the-doctype


We can call it a reference or declaration, not sure if it is technically correct. I did miss it in the actual instructions and I think we call it a declaration everywhere else, so I guess we would need to change it to be consistent. I can change it to declaration. But we might want to consider using the language as used in the specs. I can open an issue for it.

**Review by Sembauke — APPROVED** [review]

**Review by moT01 — APPROVED** [review]
LGTM @lasjorg 🎉

**Pull request merged**